src/GraphQL/Resolvers/SourceStreamMethodResolver.cs

```diff
@@ -110,20 +108,19 @@ public SourceStreamMethodResolver(MethodInfo methodInfo, LambdaExpression instan
                     var ctor = adapterType.GetConstructor(new Type[] { bodyExpression.Type })!;
                     bodyExpression = Expression.New(ctor, bodyExpression);
                 }
-                var valueTaskType = typeof(ValueTask<IObservable<object?>>);
-                var constructor = valueTaskType.GetConstructor(new Type[] { typeof(IObservable<object?>) })!;
-                return Compile(Expression.New(constructor, bodyExpression));
+                return Compile(Expression.New(_valueTaskObservableCtor, bodyExpression));
             }
+
             // IAsyncEnumerable<T>
-            else if (bodyExpression.Type.IsGenericType && bodyExpression.Type.GetGenericTypeDefinition() == typeof(IAsyncEnumerable<>))
+            if (bodyExpression.Type.IsGenericType && bodyExpression.Type.GetGenericTypeDefinition() == typeof(IAsyncEnumerable<>))
             {
                 var innerType = bodyExpression.Type.GetGenericArguments()[0];
                 var method = _convertFromAsyncEnumerableMethodInfo.MakeGenericMethod(innerType);
                 var func = method.CreateDelegate<Func<Expression, ParameterExpression, Func<IResolveFieldContext, ValueTask<IObservable<object?>>>>>();
                 return func(bodyExpression, resolveFieldContextParameter);
             }
 
-            throw new InvalidOperationException("Method must return a IObservable<T>, Task<IObservable<T>> or ValueTask<IObservable<T>> where T is a reference type, or a IAsyncEnumerable<T>, Task<IAsyncEnumerable<T>> or ValueTask<IAsyncEnumerable<T>>.");
+            throw new InvalidOperationException("Method must return a IObservable<T>, Task<IObservable<T>>, ValueTask<IObservable<T>>, IAsyncEnumerable<T>, Task<IAsyncEnumerable<T>> or ValueTask<IAsyncEnumerable<T>>.");
```

Comment:
This change is not appropriate for this PR.
```suggestion
            throw new InvalidOperationException("Method must return a IObservable<T>, Task<IObservable<T>> or ValueTask<IObservable<T>> where T is a reference type, or a IAsyncEnumerable<T>, Task<IAsyncEnumerable<T>> or ValueTask<IAsyncEnumerable<T>>.");
```

Reply:
Why? There is no more that constraint.

Reply:
There is currently.  PR #3484 has not been merged yet.

Reply:
Oh, if you're targeting my PR, then yes, it makes sense.

Reply:
Target is your branch.